source/runbooks/revoke-network-access.html.md.erb

```diff
@@ -0,0 +1,96 @@
+---
+owner_slack: "#modernisation-platform"
+title: Revoke network access if required
+last_reviewed_on: 2024-03-04
+review_in: 6 months or when is resolved.
+---
+
+# <%= current_page.data.title %>
+
+# Summary
+
+The steps listed deny access on a platform level. If additional changes are required at an application level then these must be undertaken by the application owning team.
+
+A network attack is an exceptional situation so quick amendments are needed. It is anticiapted that this can be done quickley through the console.
+
+This document lists the steps needed to stop access to and from the internet and prevening the system from being accessed from or accessing other applications on the MoJ.
+
+All changes are to be made manually on the relevant environment (for example core-vpc-production) and are not made through the code. If a way of reverting back to baseline is required the terraform code can be run and this will reset everything.
+
+## Stopping access from the internet to an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+If you know which VPC application area is being attacked you can remove it from the VPC.
+
+1. Go to VPC->Internet gateways
+2. Select the gateway (example is hmpps-preproduction-internet-gateway)
+3. Note the Internet gateway ID and the VPC ID
+4. Click on Actions, Detach from VPC
+5. If you are sure this is correct click on the Detach internet gateway button
+
+The selected internet gateway will be detached from the VPC.
+
+This change will affect all applications in the VPC that you have selected. No applications in the VPC will be accessible from the internet.
+
+## Stopping access to the internet for an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+The process needs to be done for all 4 regions listed. An example is
+
+- hmpps-production-data-nacl
+- hmpps-production-private-nacl
+- hmpps-production-protected-nacl
+- hmpps-production-public-nacl
+
+Follow the process below for each of these
+
+1. Go to VPC->Security->Network ACLs
+2. Click on the checkbox for the one you want to change then select **outbound rules**.
+3. Due to the way checks work in AWS (see [here](https://docs.aws.amazon.com/network-firewall/latest/developerguide/suricata-rule-evaluation-order.html) ) we need to amend each one in place to Deny where it is currently Allow. **Note** any that are currently set to Deny in case this is manually changed back.
+4. Save changes
+
+Setting up a deny rule on the inline-inspection firewalls, regardless of the processing order, is not possible due to the issue mentioned above. This is why the deny on each area is required.
+
+## Stopping access from the MOJ to an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
```

Comment:
Is that right? Would it not be better to block the firewall?

Reply:
Blocking at the firewall would be a lot more work than it would look like at the outset: the short version is that the `permits` all get applied before the `denies` are looked at, so you'd need to prune what you permit first before then setting the denies, or consider changing to use `alert` instead of `permit` for traffic passing through a network firewall (alert allows traffic, but logs it. permit allows traffic and doesn't log it).

Reply:
We discussed this and decided it was better to do it through this route rather than letting something bypassing here and hitting the firewall. It would also make this "quick fix" temporary measure a lot more complex.

Reply:
Fair enough @dms1981 this would be challenging (hence originally suggested changes in code), but Steve's suggestion was to amend every Allow rule to Deny, which would not only mean that we would block all the traffic (which in case of blocking MOJ internal traffic only, was not desired), but it also meant a large number of manual changes (e.g. 10 NACL rules in each subnet type, such as data, private, public, protected then multiplied by 3 regions, never mind making changes for multiple business units or multiple environments).

it's the other way around Steve, incoming traffic from MOJ, coming through the MP TGW will be first filtered through the firewall in core-network-services VPC then routed to the core-vpc-<env>

Reply:
All good points @ewastempel - I'll work with @SteveLinden to firm up his understanding here.

Reply:
Understood. I have amended the code to add a new rule, prior to the ones we want to deny, which will alleviate the need to amend all the rows.